### PR TITLE
Shapes

### PR DESCRIPTION
**Pull request by Runemoro**

### CODE REVIEW

File: mappings/net/minecraft/util/shape/VoxelShape.mapping [hunk 1]
```diff
@@ -1,7 +1,7 @@
 CLASS cri net/minecraft/util/shape/VoxelShape
-	FIELD a shape Lcqx;
+	FIELD a cellSet Lcqx;
```
Comment (Prospector) [inline mappings/net/minecraft/util/shape/VoxelShape.mapping]: `cellSet`...why not just `cells`?

Comment (Runemoro) [reply mappings/net/minecraft/util/shape/VoxelShape.mapping]: Renamed it to `voxels`.

Comment (Prospector) [reply mappings/net/minecraft/util/shape/VoxelShape.mapping]: ah yes much nicer

**Review by Prospector — APPROVED** [review]

File: mappings/net/minecraft/util/shape/DoubleListPair.mapping [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CLASS cra net/minecraft/util/shape/DoubleListPair
+	CLASS net/minecraft/util/shape/DoubleListMerger$a SectionPairPredicate
```
Comment (Gegy) [inline mappings/net/minecraft/util/shape/DoubleListPair.mapping]: The outer class name is wrong here

File: mappings/net/minecraft/util/shape/DoubleListPair.mapping [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+		METHOD merge test (III)Z
```
Comment (Gegy) [inline mappings/net/minecraft/util/shape/DoubleListPair.mapping]: Why not keep the Mojang name here?

File: mappings/net/minecraft/util/shape/DoubleListPair.mapping (hunk 2 above)
Comment (Runemoro) [reply mappings/net/minecraft/util/shape/DoubleListPair.mapping]: Fixed

File: mappings/net/minecraft/util/shape/DoubleListPair.mapping (hunk 3 above)
Comment (Runemoro) [reply mappings/net/minecraft/util/shape/DoubleListPair.mapping]: Didn't realize it was a Mojang name (using Knit so there's no highlighting). I unmapped it, though it seems weird to call the method of a predicate `merge`...

Comment (Gegy) [reply mappings/net/minecraft/util/shape/DoubleListPair.mapping]: I haven't read through the code much, but it does seem to be used specifically for merging voxel shapes.

Comment (Runemoro) [reply mappings/net/minecraft/util/shape/DoubleListPair.mapping]: It's passed as a parameter to the `forAllOverlappingSections`, which returns true if the predicate is true for all pairs of sections in both lists that overlap. It's used by both the shape comparison and shape merging code, but it still seems weird for a boolean-valued method with no side effects to be called `merge`.

Comment (Gegy) [reply mappings/net/minecraft/util/shape/DoubleListPair.mapping]: See `BitSetVoxelSet#combine`: the `SectionPairPredicate` passed to `forAllOverlappingSections` produces side effects through its implementation. Imo, `merge` does make some sense here.
Also, not necessarily a good name, but I imagine `forAllOverlappingSections` would be `merge` too

**Review by Prospector — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**